kustomize/kustomize_generator_test.go

```diff
@@ -117,3 +120,53 @@ func Test_SecureBuild_rel_basedir(t *testing.T) {
 	_, err := kustomize.SecureBuild("testdata/relbase", "testdata/relbase/clusters/staging/flux-system", false)
 	g.Expect(err).ToNot(HaveOccurred())
 }
+
+func Test_Components(t *testing.T) {
+	tests := []struct {
+		name               string
+		dir                string
+		fluxComponents     []string
+		expectedComponents []interface{}
+	}{
+		{
+			name:               "test kustomization.yaml with components and Flux Kustomization without components",
+			dir:                "components",
+			fluxComponents:     []string{},
+			expectedComponents: []interface{}{"componentA"},
+		},
+		{
+			name:               "test kustomization.yaml without components and Flux Kustomization with components",
+			dir:                "zero-components",
+			fluxComponents:     []string{"componentB", "componentC"},
+			expectedComponents: []interface{}{"componentB", "componentC"},
+		},
+		{
+			name:               "test kustomization.yaml with components and Flux Kustomization with components",
+			dir:                "components",
+			fluxComponents:     []string{"componentB", "componentC"},
+			expectedComponents: []interface{}{"componentA", "componentB", "componentC"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+			tmpDir := t.TempDir()
+			g.Expect(copy.Copy("./testdata/components", tmpDir)).To(Succeed())
```

Comment:
My understanding here is that you never copy ""./testdata/zero-components", and the `kustomization.yaml` for the second test-case is always generated. If that's correct, then do you need the test sub-repo?

Reply:
`./testdata/components` is copied and it has two directories: `components` and `zero-components`.

I have refactored the test so `zero-components` isn't needed, now it just uses the empty `tmpDir` as root if `dir` is a empty string.